libcxx/include/__tree

```diff
@@ -107,8 +111,12 @@ __root, have a non-null __parent_ field.
 // Returns:  true if __x is a left child of its parent, else false
 // Precondition:  __x != nullptr.
 template <class _NodePtr>
-inline _LIBCPP_HIDE_FROM_ABI bool __tree_is_left_child(_NodePtr __x) _NOEXCEPT {
-  return __x == __x->__parent_->__left_;
+inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX26 bool __tree_is_left_child(_NodePtr __x) _NOEXCEPT {
+  auto __0 = __x->__parent_;
+  _NodePtr __1 = __0->__left_;
+  _NodePtr __2 = __x;
+  return __1 == __2;
+  // return __x == __x->__parent_->__left_;
 }
```

Comment:
Why did this change? Also, please use better names than `0`, `1` and `2`.

Reply:
Sorry, this was a part of trying to investigate why only gcc keeps failing for the erase test case.

Will clean it up before final review

Reply:
This has been reverted as of [72f203d](https://github.com/llvm/llvm-project/pull/134330/commits/72f203dd5366e815046780fcfc2ce215782bf19a)